aws/cluster/main.tf

```diff
@@ -113,3 +113,16 @@ EOF
 
   depends_on = [null_resource.kops-cluster]
 }
+
+// Cluster Docker Login Configuration on S3
+resource "aws_s3_bucket_object" "docker-auth-config" {
+  count  = var.docker-auth-config != "" ? 1 : 0
```

Comment:
Would it make sense to make this a bit more generic solution for passing the secrets into the cluster from Terraform? We could use it for IAM/SSH keys/... The `docker-auth-config` var could be map(string) and in order to cover this case it could be simply set to 
```
{
  dockerconfig = "..."
}
```
with `{}` as defaults? It would make this variable optional and make it easy to reuse in the future.

Reply:
It _might_ be useful but I don't think it's worth doing right now, delaying this change further and introducing further changes in various other places in these modules which are not linked to our current focus.

Reply:
The sad thing is, that making this solution generic right from the beginning would not have cost us anything - there will be literally no added code, no added testing complexity, not anything against it, but it would have increased reusability for us, potentially saving the time in the future. And given this is a public repo, I believe others would benefit from generic approach too and it is a good motivation not to write the code based on our current needs only. Especially if it would have had cost us nothing if it was implemented right from the beginning...

Reply:
Thanks for your thoughts.
Unfortunately, nothing is free - it would still cost us to actually move to use it at a time when we don't really plan to do such work. I don't think we'll do any more work here until we move to 1.15 anyway. So I see this as an improvement for Karch 1.15 not really at this time or for this case in particular.

I don't think we have any data as to who uses this repo, so as I said it's only a slight chance others would benefit.

If I get a chance to test and refactor today then I'll see what I can do otherwise it's a good proposal for the Karch 1.15 update. Then we can actually benefit in various places and perform the migration and changes to benefit from such a method.

We welcome PRs or tickets should people be interested in this type of change.